overlord/snapstate/backend/extra_info.go

```diff
@@ -0,0 +1,113 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package backend
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+var (
+	ErrExtraInfoExtraContent = errors.New("extra content found after extra info")
+	ErrNoExtraInfo           = errors.New("extra info not found")
+)
+
+// ExtraInfo is information about a snap that is not in the snap.yaml,
+// not needed in the state, but may be cached to augment the
+// information returned for locally-installed snaps
+type ExtraInfo struct {
+	Media snap.MediaInfos `json:"media,omitempty"`
+}
+
+func extraFilename(snapName string) string {
+	return filepath.Join(dirs.SnapExtraInfoDir, snapName) + ".json"
+}
+
+func LoadExtraInfo(info *snap.Info) error {
```

Comment:
Nitpick: I was expecting this to return the extra info, perhaps `AttachExtraInfo` or something similar?